aes-gcm/src/lib.rs

```diff
@@ -280,9 +280,9 @@ where
 
         // TODO(tarcieri): interleave encryption with GHASH
         // See: <https://github.com/RustCrypto/AEADs/issues/74>
-        ctr.apply_keystream_partial(buffer.into());
+        ctr.apply_keystream_partial(buffer.reborrow());
 
-        let full_tag = self.compute_tag(mask, associated_data, buffer);
+        let full_tag = self.compute_tag(mask, associated_data, buffer.get_in());
```

Comment:
This one probably needs to be `get_out`, to compute a MAC over the ciphertext

Reply:
Yeah there is probably a whole slew of those, I assumed the backing buffer was unique.